scipy/optimize/_differentialevolution.py

```diff
@@ -480,41 +622,85 @@ def solve(self):
 
         # do the optimisation.
         for nit in range(1, self.maxiter + 1):
+        
             if self.dither is not None:
                 self.scale = self.random_number_generator.rand(
                 ) * (self.dither[1] - self.dither[0]) + self.dither[0]
-            for candidate in range(np.size(self.population, 0)):
+                
+            # brake the population to subpopulation depending on the size of the pool
+            # determine number of sub-populations 'nsp', or number of 
+            # parallel evaluations, which is 'nsp + 1'
+            # and the length of the reminder 'lenrem'
+            nsp, lenrem = divmod(np.size(self.population, 0), self.poolsize)
+            
+            itsp = 0
+            # iterate among sub-populations
+            # when self.poolsize == 1 the mutation is aggressive
+            # when self.poolsize > 1 the mutation is quasi-aggressive
+            while itsp <= nsp:
+                # determine the length of the subpopulation
+                lensp = self.poolsize if (itsp < nsp) else lenrem
+
+                # initialize list for all members (parameters) of the current subpopulation
+                spparams = []
```

Comment:
Replace lines 645 through to 652 by:

    # one could vectorize self._mutate, but I'm not sure if it's worth it.
    trials = np.array(map(self._mutate, range(candidate, candidate + self.poolsize * itsp))

    # ensure constraints
    trials[trials < 0.] = np.random.random(np.count_nonzero(trials < 0))
    trials[trials > 1.] = np.random.random(np.count_nonzero(trials > 1.))

    # scale trials
    params = self._scale_parameters(trials)
    nfev += lensp